lib/cell.rb

```diff
@@ -4,22 +4,43 @@ class Cell
   def initialize(coordinate)
     @coordinate = coordinate
     @ship = nil
+    @fired_upon = false
   end
 
   def empty?
-    @ship.nil?
```

Comment:
@ship.nil? does the same thing as if @ship == nil return true, else false. (its just a shorter way of writing it!)